src/ssa.c

```diff
@@ -21,6 +21,45 @@
 /* Dead store elimination window size */
 #define OVERWRITE_WINDOW 3
 
+void var_list_ensure_capacity(var_list_t *list, int min_capacity)
+{
+    if (list->capacity >= min_capacity) {
+        return;
+    }
+
+    int new_capacity = list->capacity ? list->capacity : 8;
+
+    while (new_capacity < min_capacity) {
+        new_capacity <<= 1;
+    }
+
+    var_t **new_elements = arena_alloc(BB_ARENA, new_capacity * HOST_PTR_SIZE);
+
+    if (list->elements)
+        memcpy(new_elements, list->elements, list->size * HOST_PTR_SIZE);
```

Comment:
Since the new code might introduce several additional memory copy operations, I imagine it will be slower. I'm not sure what the performance impact will be, but I suspect the 75% memory saving is very likely worth that cost.

However, this is a trade-off. Reading the commit message alone gives the impression that this is a pure improvement without any drawbacks.

I'm wondering if this performance difference has been measured?